### ISSUE TITLE
Multiple services, one ESP

### ISSUE THREAD

**Issue by sgammon** [GoogleCloudPlatform/esp-v2#434]
Is it possible? I realize one can mount multiple services in the API yaml, but composing APIs in this way requires a different service to be published in, for instance, the Service Control API.

I remember there was previously functionality for this in ESPv1, but it may have been experimental (and may have been eliminated).

**Comment by TAOXUY** [comment]
Yes, having multiple services defined in the API yaml is totally possible. We even have integration test for [that](https://github.com/GoogleCloudPlatform/esp-v2/blob/a8f86f24e01715a83c5c959eb953bb19e4e19d8f/tests/integration_test/multi_grpc_services_test.go#L30) and I have e2e tested that locally before.

**Comment by TAOXUY** [comment]
IMO, it is worth to make multi-services config example under `/examples`. Will do it later.

**Comment by qiwzhang** [comment]
Just some clarification; word `service` may mean different things for different people. 

1) a gRPC service;  a server (running in one host:port or an endpoint) may support multiple gRPC services. Yes,  ESPv2 supports multiple gRPC services that the backend supports.  All these services are routed to the same backend endpoint.  You just specify them in [Apis](https://github.com/googleapis/googleapis/blob/master/google/api/service.proto#L99) in the endpoint service config yam file. 

2) a service means an endpoint, or a server identified by host:port.   Yes, we supports multiple backend endpoints too.  For http, in openapi config,  you can use `address` inside `x-google-backend` to specify different backend endpoints for different paths.  For gRpc config,  you may have to use Backend rule with wildcard selector to point to different backend endpoints.

For example: you have backend service A with address_A,  and service B with address_B.
*  In the [apis](https://github.com/googleapis/googleapis/blob/master/google/api/service.proto#L99), you have to list both A and B
*  In the BackendRule [rules](https://github.com/googleapis/googleapis/blob/master/google/api/backend.proto#L30), you can do
```
Backend
- rules
  - selector: /service_A/*
    address: address_A
  - selector: /service_B/*
    address: address_B
```
